### PR TITLE
Python: add ZDIFF command

### PR DESCRIPTION
**Pull request by aaron-congo**
https://redis.io/docs/latest/commands/zdiff/
Note: depends on update to cluster_routing.rs in redis-rs

### CODE REVIEW

File: python/python/glide/async_commands/core.py [hunk 1]
```diff
@@ -2326,6 +2326,58 @@ async def zscore(self, key: str, member: str) -> Optional[float]:
             await self._execute_command(RequestType.ZScore, [key, member]),
         )
 
+    async def zdiff(self, keys: List[str]) -> List[str]:
+        """
+        Returns the difference between the first sorted set and all the successive sorted sets.
+        To get the elements with their scores, see `zdiff_withscores`.
+
+        See https://redis.io/commands/zdiff/ for more details.
+
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:2335]: please add same slot notice

File: python/python/tests/test_async_client.py [hunk 2]
```diff
@@ -1944,6 +1944,43 @@ async def test_zrank(self, redis_client: TRedisClient):
...
+        assert await redis_client.zdiff([key1, key3]) == []
+        assert await redis_client.zdiff([non_existing_key, key3]) == []
+
+        assert await redis_client.zdiff_withscores([key1, key2]) == {
+            "one": 1.0,
+            "three": 3.0,
+        }
+        assert await redis_client.zdiff_withscores([key1, key3]) == {}
+        assert await redis_client.zdiff_withscores([non_existing_key, key3]) == {}
+
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_async_client.py:1974]: please add a test for cross-slot case

File: python/python/glide/async_commands/core.py (hunk 1 above)
Comment (aaron-congo) [reply python/python/glide/async_commands/core.py:2336]: Good point, note that his notice is missing for the sorted set commands in Java so they will need to be updated

File: python/python/tests/test_async_client.py (hunk 2 above)
Comment (aaron-congo) [reply python/python/tests/test_async_client.py:1974]: Good point, note that this test is missing for the sorted set commands in Java so they will need to be updated

File: python/python/glide/async_commands/core.py (hunk 1 above)
Comment (Yury-Fridlyand) [reply python/python/glide/async_commands/core.py:2336]: See #226

File: python/python/glide/async_commands/core.py [hunk 3]
```diff
@@ -2326,6 +2326,59 @@ async def zscore(self, key: str, member: str) -> Optional[float]:
...
+        Returns the difference between the first sorted set and all the successive sorted sets.
+        When in Cluster mode, all keys must map to the same hash slot.
+
+        See https://valkey.io/commands/zdiff for more details.
+
+        Args:
+            keys (List[str]): The keys of the sorted sets.
+
+        Returns:
+            Dict[str, float]: A dictionary of elements and their scores representing the difference between the sorted
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:2366]: `Dict` or `Mapping`?

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
Maybe you need to rebase your branches to let CI pass

**Review by GumpacG — APPROVED** [review]

File: python/python/glide/async_commands/core.py (hunk 3 above)
Comment (aaron-congo) [reply python/python/glide/async_commands/core.py:2760]: I wasn't aware of the Mapping type but apparently it is a more general type than Dict. Its unclear which one is better, the existing code uses both somewhat interchangeably. According to [this SO post](https://stackoverflow.com/a/75722246/22506474) Dict is preferred for return types and Mapping for arguments, but the existing code has both types in both places

**Pull request merged**